pkg/snapstore/s3_snapstore.go

```diff
@@ -495,30 +496,52 @@ func (s *S3SnapStore) Delete(snap brtypes.Snapshot) error {
 	return err
 }
 
-// S3SnapStoreHash calculates and returns the hash of aws S3 snapstore secret.
-func S3SnapStoreHash(config *brtypes.SnapstoreConfig) (string, error) {
+// S3SnapStoreModTime returns the lastest modification timestamp of the AWS credential file(s)
+func S3SnapStoreModTime(config *brtypes.SnapstoreConfig) (time.Time, error) {
 	if dir, isSet := os.LookupEnv(awsCredentialFile); isSet {
-		awsConfig, err := readAWSCredentialFromDir(dir)
+		awsTimeStamp, err := readAWSCredentialModTime(dir)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials from %v directory", dir)
+			return time.Time{}, fmt.Errorf("failed to get AWS credential timestamp from the directory %v with error: %v", dir, err)
 		}
-		return getS3Hash(awsConfig), nil
+		return awsTimeStamp, nil
 	}
 
 	if filename, isSet := os.LookupEnv(awsCredentialJSONFile); isSet {
-		awsConfig, err := credentialsFromJSON(filename)
+		fileInfo, err := os.Stat(filename)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials using %v file", filename)
+			return time.Time{}, fmt.Errorf("failed to fetch file information of the AWS JSON credential file %v with error: %v", filename, err)
 		}
-		return getS3Hash(awsConfig), nil
+		return fileInfo.ModTime(), nil
 	}
 
-	return "", nil
+	return time.Time{}, fmt.Errorf("No environment variable set for the AWS credential file")
 }
 
-func getS3Hash(config *awsCredentials) string {
-	data := fmt.Sprintf("%s%s%s", config.AccessKeyID, config.SecretAccessKey, config.Region)
-	return getHash(data)
+// readAWSCredentialModTime reads the modification timestamp of each AWS credential file and returns the latest
+func readAWSCredentialModTime(dirname string) (time.Time, error) {
```

Comment:
There is too much of code repetition across different providers. Please create a single utility function which given a set of paths, it will get the latest modified time amongst the files.

Reply:
Addressed in the [commit](https://github.com/gardener/etcd-backup-restore/pull/670/commits/351821d5fa413798bb159a1ed6ce411ae086a2ae)